nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/test/java/org/apache/nifi/processors/standard/AbstractTestConversion.java

```diff
@@ -0,0 +1,396 @@
+            commonReaderConfiguration(testRunner);
+        };
+    }
+
+    protected void fromXml(String postfix) {
+        reader = new XMLReader();
+        inputHandler = stringInputHandler(getContent(postfix));
+
+        readerConfigurer = testRunner -> {
+            commonReaderConfiguration(testRunner);
+            testRunner.setProperty(reader, XMLReader.RECORD_FORMAT, XMLReader.RECORD_ARRAY);
+        };
+    }
+
+    protected void fromAvro(String postfix) {
+        reader = new AvroReader();
+        inputHandler = byteInputHandler(getByteContent(postfix));
+
+        readerConfigurer = testRunner -> {
+            commonReaderConfiguration(testRunner);
+        };
+    }
+
+    protected void toCsv(String postfix) {
+        writer = new CSVRecordSetWriter();
+        resultHandler = stringOutputHandler(getContent(postfix));
+
+        writerConfigurer = testRunner -> {
+            commonWriterConfiguration(testRunner);
+        };
+    }
+
+    protected void toJson(String postfix) {
+        writer = new JsonRecordSetWriter();
+        resultHandler = stringOutputHandler(getContent(postfix));
+
+        writerConfigurer = testRunner -> {
+            commonWriterConfiguration(testRunner);
+            testRunner.setProperty(writer, "Pretty Print JSON", "true");
+        };
+    }
+
+    protected void toXml(String postfix) {
+        writer = new XMLRecordSetWriter();
+        resultHandler = stringOutputHandler(getContent(postfix));
+
+        writerConfigurer = testRunner -> {
+            commonWriterConfiguration(testRunner);
+            testRunner.setProperty(writer, "pretty_print_xml", "true");
+            testRunner.setProperty(writer, "root_tag_name", "root");
+            testRunner.setProperty(writer, "record_tag_name", "nifiRecord");
+        };
+    }
+
+    protected void toAvro(String postfix) {
+        writer = new AvroRecordSetWriter();
+        resultHandler = mockFlowFile -> {
+            try {
+                ArrayList<Map<String, Object>> expected = getRecords(getByteContent(postfix));
+                ArrayList<Map<String, Object>> actual = getRecords(mockFlowFile.toByteArray());
+
+                assertEquals(expected, actual);
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        };
+
+        writerConfigurer = testRunner -> {
+            commonWriterConfiguration(testRunner);
+        };
+    }
+
+    protected Consumer<TestRunner> stringInputHandler(String input) {
+        return testRunner -> testRunner.enqueue(input);
+    }
+
+    protected Consumer<TestRunner> byteInputHandler(byte[] input) {
+        return testRunner -> testRunner.enqueue(input);
+    }
+
+    protected Consumer<MockFlowFile> stringOutputHandler(String expected) {
+        return mockFlowFile -> mockFlowFile.assertContentEquals(expected);
+    }
+
+    protected String getContent(String postfix) {
+        String content = new String(getByteContent(postfix));
```

Comment:
We can simply return `new String(getByteContent(posfix));`

Reply:
Yeah, I guess here half my previous reasoning is moot so in such simple cases it's fine to have a single line that returns an expression.